src/fides/api/api/v1/endpoints/dataset_config_endpoints.py

```diff
@@ -699,87 +509,6 @@ def get_ctl_datasets(
     return datasets
 
 
-def recursive_clean_fields(fields: List[dict]) -> List[dict]:
-    """
-    Recursively clean the fields of a dataset.
-    """
-    cleaned_fields = []
-    for field in fields:
-        field["name"] = field["name"].split(".")[-1]
-        if field["fields"]:
-            field["fields"] = recursive_clean_fields(field["fields"])
-        cleaned_fields.append(field)
-    return cleaned_fields
-
-
-def run_clean_datasets(
-    db: Session, datasets: List[Dataset]
-) -> tuple[List[str], List[str]]:
-    """
-    Clean the dataset name and structure to remove any malformed data possibly present from nested field regressions.
-    Changes dot separated positional names to source names (ie. `user.address.street` -> `street`).
-    """
-
-    for dataset in datasets:
-        logger.info(f"Cleaning field names for dataset: {dataset.fides_key}")
-        for collection in dataset.collections:
-            collection["fields"] = recursive_clean_fields(collection["fields"])  # type: ignore # pylint: disable=unsupported-assignment-operation
-
-        # manually upsert the dataset
-
-        logger.info(f"Upserting dataset: {dataset.fides_key}")
-        failed = []
-        try:
-            dataset_ctl_obj = (
-                db.query(CtlDataset)
-                .filter(CtlDataset.fides_key == dataset.fides_key)
-                .first()
-            )
-            if dataset_ctl_obj:
-                db.query(CtlDataset).filter(
-                    CtlDataset.fides_key == dataset.fides_key
-                ).update(
-                    {
-                        "collections": dataset.collections,
-                        "updated_at": datetime.now(timezone.utc),
-                    },
-                    synchronize_session=False,
-                )
-                db.commit()
-            else:
-                logger.error(f"Dataset with fides_key {dataset.fides_key} not found.")
-        except Exception as e:
-            logger.error(f"Error upserting dataset: {dataset.fides_key} {e}")
-            db.rollback()
-            failed.append(dataset.fides_key)
-
-    succeeded = [dataset.fides_key for dataset in datasets]
-    return succeeded, failed
-
-
-@router.get(
-    "/datasets/clean",
-    dependencies=[Security(verify_oauth_client, scopes=[DATASET_READ])],
-    response_model=List[Dataset],
-    deprecated=True,
-)
-def clean_datasets(
```

Comment:
I moved this to the `DatasetService` since this has to do with standalone datasets